Files/ViewModels/ItemViewModel.cs

```diff
@@ -753,48 +753,39 @@ private async Task LoadItemThumbnail(ListedItem item, uint thumbnailSize = 20, I
-                        item.FileImage = await item.CustomIconData.ToBitmapAsync();
-                        item.LoadUnknownTypeGlyph = false;
-                        item.LoadWebShortcutGlyph = false;
-                        item.LoadFileIcon = true;
-                    }, Windows.System.DispatcherQueuePriority.Low);
-                    wasIconLoaded = true;
-                }
-                else
+                if (!item.IsShortcutItem && !item.IsHiddenItem && !FtpHelpers.IsFtpPath(item.ItemPath))
                 {
-                    if (!item.IsShortcutItem && !item.IsHiddenItem && !FtpHelpers.IsFtpPath(item.ItemPath))
+                    var matchingStorageFile = matchingStorageItem.AsBaseStorageFile() ?? await GetFileFromPathAsync(item.ItemPath);
+                    if (matchingStorageFile != null)
                     {
-                        var matchingStorageFile = matchingStorageItem.AsBaseStorageFile() ?? await GetFileFromPathAsync(item.ItemPath);
-                        if (matchingStorageFile != null)
-                        {
-                            var mode = thumbnailSize < 80 ? ThumbnailMode.ListView : ThumbnailMode.SingleItem;
+                        var mode = thumbnailSize < 80 ? ThumbnailMode.ListView : ThumbnailMode.SingleItem;
 
-                            using var Thumbnail = await matchingStorageFile.GetThumbnailAsync(mode, thumbnailSize, ThumbnailOptions.ResizeThumbnail);
-                            if (!(Thumbnail == null || Thumbnail.Size == 0 || Thumbnail.OriginalHeight == 0 || Thumbnail.OriginalWidth == 0))
+                        using var Thumbnail = await matchingStorageFile.GetThumbnailAsync(mode, thumbnailSize, ThumbnailOptions.ResizeThumbnail);
+                        if (!(Thumbnail == null || Thumbnail.Size == 0 || Thumbnail.OriginalHeight == 0 || Thumbnail.OriginalWidth == 0))
+                        {
+                            await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(async () =>
                             {
-                                await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(async () =>
-                                {
-                                    item.CustomIconData = await Thumbnail.ToByteArrayAsync();
-                                    item.FileImage = await item.CustomIconData.ToBitmapAsync();
-                                    item.LoadUnknownTypeGlyph = false;
-                                    item.LoadWebShortcutGlyph = false;
-                                    item.LoadFileIcon = true;
-                                }, Windows.System.DispatcherQueuePriority.Low);
-                                wasIconLoaded = true;
-                            }
+                                item.LoadUnknownTypeGlyph = false;
+                                item.LoadWebShortcutGlyph = false;
+                                item.LoadFolderGlyph = false;
+                                item.LoadFileIcon = true;
```

Comment:
Ensures Image is added to visual tree before we try to set a source as per [this](https://docs.microsoft.com/en-us/windows/uwp/debug-test-perf/app-analysis#image-is-being-called-when-the-imagesource-is-not-in-the-live-tree) recommendation